# This file is part of HiGFXback

# requires
REQUIRES="lxml-build make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix /fb

# build
make -C api

# install
install -d $DESTDIR/fb/include/EGL
install -m 644 api/EGL/egl.h $DESTDIR/fb/include/EGL
install -m 644 api/EGL/eglext.h $DESTDIR/fb/include/EGL
install -m 644 api/EGL/eglplatform.h $DESTDIR/fb/include/EGL
install -d $DESTDIR/fb/include/KHR
install -m 644 api/KHR/khrplatform.h $DESTDIR/fb/include/KHR
install -d $DESTDIR/fb/lib/pkgconfig
install -m 644 egl-registry.pc $DESTDIR/fb/lib/pkgconfig
install -d $DESTDIR/fb/share/egl/registry
install -m 644 api/egl.xml $DESTDIR/fb/share/egl/registry

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/egl-registry-build.pc << EOF
Name: egl-registry
Version: 20170204
Description: EGL API Registry
Requires: $REQUIRES

devel=\\
/fb/include/EGL/egl.h \\
/fb/include/EGL/eglext.h \\
/fb/include/EGL/eglplatform.h \\
/fb/include/KHR/khrplatform.h \\
/fb/lib/pkgconfig/egl-registry.pc \\
/fb/share/egl/registry/egl.xml
EOF

exit
--- EGL-Registry-20170204.orig/Configure
+++ EGL-Registry-20170204/Configure
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while : ; do
+  case "$1" in
+    "")
+      break
+    ;;
+    --prefix)
+      PREFIX=$2
+      shift
+    ;;
+    *)
+      echo Unknown option \"$1\".
+      exit 1
+    ;;
+  esac
+  shift
+done
+
+sed "/^prefix/s|=.*|=$PREFIX|" api/egl-registry.pc.in > egl-registry.pc
--- EGL-Registry-20170204.orig/api/EGL/eglplatform.h
+++ EGL-Registry-20170204/api/EGL/eglplatform.h
@@ -66,8 +66,13 @@
  * portability of applications using EGL running on different EGL
  * implementations.
  */
+#if defined(EGL_NO_PLATFORM_SPECIFIC_TYPES)
 
-#if defined(_WIN32) || defined(__VC32__) && !defined(__CYGWIN__) && !defined(__SCITECH_SNAP__) /* Win32 and WinCE */
+typedef void *EGLNativeDisplayType;
+typedef void *EGLNativePixmapType;
+typedef void *EGLNativeWindowType;
+
+#elif defined(_WIN32) || defined(__VC32__) && !defined(__CYGWIN__) && !defined(__SCITECH_SNAP__) /* Win32 and WinCE */
 #ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN 1
 #endif
--- EGL-Registry-20170204.orig/api/Makefile
+++ EGL-Registry-20170204/api/Makefile
@@ -16,7 +16,7 @@
 EGLHEADERS = EGL/egl.h EGL/eglext.h
 
 # Generation tools
-PYTHON = python
+PYTHON = python3
 PYFILES = genheaders.py reg.py
 REGISTRY = egl.xml
 GENOPTS =
--- EGL-Registry-20170204.orig/api/egl-registry.pc.in
+++ EGL-Registry-20170204/api/egl-registry.pc.in
@@ -0,0 +1,8 @@
+prefix=@prefix@
+datarootdir=${prefix}/share
+headersdir=${pc_sysrootdir}${prefix}/include
+pkgdatadir=${pc_sysrootdir}${datarootdir}/egl/registry
+
+Name: EGL Registry
+Description: EGL API registry file
+Version: 20170204
--- EGL-Registry-20170204.orig/api/genheaders.py
+++ EGL-Registry-20170204/api/genheaders.py
@@ -75,10 +75,10 @@
 startTime = None
 def startTimer():
     global startTime
-    startTime = time.clock()
+    startTime = time.process_time()
 def endTimer(msg):
     global startTime
-    endTime = time.clock()
+    endTime = time.process_time()
     if (timeit):
         write(msg, endTime - startTime)
         startTime = None
@@ -156,8 +156,8 @@
     '** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.',
     '*/',
     '/*',
-    '** This header is generated from the Khronos OpenGL / OpenGL ES XML',
-    '** API Registry. The current version of the Registry, generator scripts',
+    '** This header is generated from the Khronos EGL XML API Registry.',
+    '** The current version of the Registry, generator scripts',
     '** used to make the header, and the header can be found at',
     '**   http://www.khronos.org/registry/egl',
     '**',
